### PR TITLE
Initial skeleton inventory

### PR DESCRIPTION
**Pull request by slwilliams**
Not functional, nor has any 3D model functionality yet. Just the initial structure for opening, closing and selecting a slot.

**Preview here:** https://pr-425.lba2remake.net

### CODE REVIEW

File: src/ui/GameUI.tsx [hunk 1]
```diff
@@ -100,7 +101,8 @@ export default class GameUI extends React.Component<GameUIProps, GameUIState> {
                 uiState.ask.choices.length === 0 &&
                 uiState.text === null &&
                 uiState.foundObject === null &&
-                !(uiState.showMenu || uiState.inGameMenu);
+                !(uiState.showMenu || uiState.inGameMenu) &&
+                !this.props.uiState.inventory;
```
Comment (agrande) [inline src/ui/GameUI.tsx:105]: You can remove this.props here

File: src/ui/GameUI.tsx [hunk 2]
```diff
@@ -109,27 +111,40 @@ export default class GameUI extends React.Component<GameUIProps, GameUIState> {
                 }
                 game.pause(false);
             }
+            const showInventory =
+                !uiState.loading &&
+                uiState.ask.choices.length === 0 &&
+                uiState.text === null &&
+                uiState.foundObject === null &&
+                !(uiState.showMenu || uiState.inGameMenu) &&
+                !this.props.uiState.behaviourMenu;
```
Comment (agrande) [inline src/ui/GameUI.tsx:120]: Same, this.props is not necessary.

File: src/ui/GameUI.tsx [hunk 3]
```diff
@@ -109,27 +111,40 @@ export default class GameUI extends React.Component<GameUIProps, GameUIState> {
...
-            uiState,
             game,
         } = this.props;
-        const hideBehaviourMenu =
-            !uiState.loading &&
-            uiState.ask.choices.length === 0 &&
-            uiState.text === null &&
-            uiState.foundObject === null &&
-            !(uiState.showMenu || uiState.inGameMenu);
-        if (hideBehaviourMenu && this.isBehaviourKey(key)) {
```
Comment (agrande) [inline src/ui/GameUI.tsx:127]: Why did you remove this block?

File: src/ui/game/Inventory.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                break;
+            case 40:
+            case 'ArrowDown':
+                if (selectedSlot > inventoryColumns * inventoryRows - inventoryColumns) {
+                    newSlot = selectedSlot - inventoryColumns * (inventoryRows - 1);
+                } else {
+                    newSlot = selectedSlot + inventoryColumns;
+                }
+                setSelectedSlot(newSlot);
+                game.getState().hero.inventorySlot = newSlot;
```
Comment (agrande) [inline src/ui/game/Inventory.tsx:57]: This part is similar in all cases you can probably move it down to the if (action) part bellow.

File: src/ui/game/Inventory.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        };
+    });
+
+    const inventorySlots = [];
+    for (let i = 0; i < inventoryRows; i += 1) {
+        for (let j = 0; j < inventoryColumns; j += 1) {
+            const slot = i * inventoryColumns + j;
+            inventorySlots.push(
+                <div
+                  key={String(slot)}
```
Comment (agrande) [inline src/ui/game/Inventory.tsx:80]: Is it required to explicitly cast it as string?

**Review by agrande — COMMENTED** [review]
Lgtm overall, just a few minor comments.

File: src/ui/GameUI.tsx (hunk 3 above)
Comment (slwilliams) [reply src/ui/GameUI.tsx:127]: it's unnecessary, if the behaviour menu is already open non of these conditions can be true. There can't be any case where the menu is open and the user takes there finger off the ctrl key and it shouldn't close.

File: src/ui/game/Inventory.tsx (hunk 4 above)
Comment (slwilliams) [reply src/ui/game/Inventory.tsx:57]: ah yea good point

File: src/ui/game/Inventory.tsx (hunk 5 above)
Comment (slwilliams) [reply src/ui/game/Inventory.tsx:80]: apparently not, removed.

File: src/ui/game/Inventory.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    });
+
+    const inventorySlots = [];
+    for (let i = 0; i < inventoryRows; i += 1) {
+        for (let j = 0; j < inventoryColumns; j += 1) {
+            const slot = i * inventoryColumns + j;
+            inventorySlots.push(
+                <div
+                  key={slot}
+                  id={String(slot)}
```
Comment (agrande) [inline src/ui/game/Inventory.tsx:75]: Same question as with the key

Comment (slwilliams) [reply src/ui/game/Inventory.tsx:75]: oh right no id has to be a string.

Comment (agrande) [reply src/ui/game/Inventory.tsx:75]: Why do you need an id btw?

**Comment by agrande** [comment]
After testing I found an issue when scrolling down with the down arrow key on the first column, it doesn't loop properly.
Also, the slots are rectangular when the aspect ratio is not the same as the original, but I suppose that can be improved in a future PR.

File: src/ui/game/Inventory.tsx (hunk 6 above)
Comment (slwilliams) [reply src/ui/game/Inventory.tsx:75]: hmm yea that's a good point, removed for now.

**Comment by slwilliams** [comment]
Fixed the boundary condition for the scrolling, yea the aspect ratio thing is interesting can take a look in a follow up PR.

**Review by agrande — APPROVED** [review]
